FreeRTOS/FreeRTOS-Kernel
# GCC: ARM_CM0: Fix L6286E error on Keil MDK

Keil MDK Compilation for CM0 core reports L6286E: Relocation #REL:0 in portasm.o

Description
-----------
Change the .b instruction to .bl with higher range to solve error reported by MDK descibed bellow.

Fix:
Error: L6286E: Relocation #REL:0 in portasm.o(.text.SVC_Handler) with respect to vPortSVCHandler_C. Value(0x1a04) out of range(-0x800 - 0x7fe) for (R_ARM_THM_JUMP11)

Compiler: Keil MDK ARMClang 6.22.0

https://developer.arm.com/documentation/ka002847/latest/ https://developer.arm.com/documentation/dui0496/m/Linker-Errors-and-Warnings/List-of-the-armlink-error-and-warning-messages

See the issue #1130 

Test Steps
-----------
See the issue #1130 

To Reproduce
git clone https://github.com/TomasGalbickaNXP/freertos-port-cm0-core-issue.git
Checkout the Example with added CM0 MPU port.
git checkout 40af92d06408450e11ca10ad17cd578760fd861a
Open MDK Project and build. It will fail with error reported above.
Checkout fix
git checkout 72a9d6246277ae32594bab69fef66c6e13af3cd1
Open MDK Project and build.
The build reports no Errors.

Checklist:
----------
- [ x ] I have tested my changes. No regression in existing tests.
- [    ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
#1130 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
